### PR TITLE
test: Refactor Smoke test

### LINKED ISSUE #952: Refactor smoke test
At the moment, our smoke test is using a different testing framework `e2e-framework` compare to `envtest` we are using for e2e test.

The workload to write a good test case with e2e-framework is significantly higher compared to `envtest`, which also increases our maintenance effort for keep two different test framework. 

Some of the test cases are also well covered in e2e, for example: "kyma delete", not make sense keep it redundant. 

AC:
- [x] Keep Kustomize dry-run in smoke test
- [x] Keep "Deploy template operator module template and enable module"  in smoke test, this is enough to verify LM works with single cluster mode (https://github.com/kyma-project/lifecycle-manager/blob/1e76346417e46f74279e76c7d01d0d5e815c876c/.github/workflows/test-smoke.yml#L127)
- [x] ~~Remove "control-plane" as target in `cli-integration` job, smoke test only verify default single cluster mode~~
- [x] Rewrite and migrate "module upgrade" test to e2e test using envtest
- [x] Delete "kyma delete" test
- [x] Smoke test directory should be gone
- [x] Drop `e2e-framework` in LM dependencies
- [ ] Update tide configuration (https://github.com/kyma-project/test-infra/pull/9178/files ), incl. `E2E (module-upgrade)` or name of new E2E test

### PR DESCRIPTION
**Pull request by nesmabadr**
**Description**

Changes proposed in this pull request:

- Remove smoke test folder
- Adjust smoke test GitHub action to only use single cluster
- Move module upgrade test to e2e test

**Related issue(s)**
Resolves #952

### CODE REVIEW

File: .github/workflows/test-e2e.yml [hunk 1]
```diff
@@ -155,7 +155,7 @@ jobs:
           done
       - name: Checkout template-operator
         uses: actions/checkout@v3
-        if: ${{ matrix.e2e-test != 'watcher-enqueue' && matrix.e2e-test != 'kyma-metrics'}}
+        if: ${{ matrix.e2e-test != 'watcher-enqueue' && matrix.e2e-test != 'kyma-metrics' && matrix.e2e-test != 'module-upgrade'}}
```
Comment (ruanxin) [inline .github/workflows/test-e2e.yml:158]: let's remove this if condition, not make too much sense maintain this template-operator checkout, we can allow all tests have this checkout step, even not in use later.

File: .github/workflows/test-smoke.yml [hunk 2]
```diff
@@ -113,19 +113,18 @@ jobs:
...
           chmod +x istio-$ISTIO_VERSION/bin/istioctl
           mv istio-$ISTIO_VERSION/bin/istioctl /usr/local/bin
           istioctl install --set profile=demo -y
           
           kubectl create namespace kyma-system
           kubectl create namespace kcp-system
-
           kubectl apply -f https://raw.githubusercontent.com/prometheus-community/helm-charts/kube-prometheus-stack-47.0.0/charts/kube-prometheus-stack/crds/crd-servicemonitors.yaml
-      - run: kyma --ci alpha deploy -k config/${{ matrix.target }}
+      - name: Deploy lifecycle manager
```
Comment (ruanxin) [inline .github/workflows/test-smoke.yml:124]:
```suggestion
      - name: Deploy lifecycle manager with Kyma CLI
```

File: go.mod [hunk 3]
```diff
@@ -20,10 +20,9 @@ require (
 	go.uber.org/zap v1.26.0
 	golang.org/x/sync v0.4.0
 	golang.org/x/time v0.3.0
-	gopkg.in/yaml.v3 v3.0.1
+	gopkg.in/yaml.v3 v3.0.1 // indirect
```
Comment (ruanxin) [inline go.mod:23]: if this is indrect, relocate it into indirect require block

File: pkg/testutils/modulecr.go [hunk 4]
```diff
@@ -126,3 +128,45 @@ func ModuleCRIsInExpectedState(ctx context.Context,
...
+	var deployment appsv1.Deployment
+	err := clnt.Get(ctx, client.ObjectKey{
+		Namespace: namespace,
+		Name:      deploymentName,
+	}, &deployment)
+
+	return err == nil && deployment.Status.AvailableReplicas != 0
+}
+
+func ChangeKymaModuleChannel(ctx context.Context,
```
Comment (ruanxin) [inline pkg/testutils/modulecr.go:146]: This function better put it in /testutils/kyma.go, and if you put there, you can find there is already similar one exists, `UpdateKymaModuleChannel`

File: tests/e2e_test/module_upgrade_test.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+		})
+
+		It("When Template Operator channel is changed", func() {
+			Eventually(UpdateKymaModuleChannel).
+				WithContext(ctx).
+				WithArguments(runtimeClient, defaultRemoteKymaName, remoteNamespace, "fast").
+				Should(Succeed())
+		})
+
+		It("Then new template operator manager exists", func() {
```
Comment (ruanxin) [inline tests/e2e_test/module_upgrade_test.go:55]: here it's better check again `"Then module CR exists"`

Comment (ruanxin) [reply tests/e2e_test/module_upgrade_test.go:55]: also the `"template-operator-v1-controller-manager"` should not exists

File: tests/e2e_test/module_upgrade_test.go [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+		})
+
+		It("And new template operator manager exists", func() {
+			Eventually(ModuleDeploymentExists).
+				WithContext(ctx).
+				WithArguments(runtimeClient, "template-operator-system", "template-operator-v1-controller-manager").
+				Should(BeFalse())
+		})
+
+		It("And old template operator manager does not exist", func() {
```
Comment (ruanxin) [inline tests/e2e_test/module_upgrade_test.go:69]: the new and old operator name should be exchanged, I hope the test will fail.

**Review by ruanxin — CHANGES_REQUESTED** [review]
only one comment.

**Review by ruanxin — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]

**Pull request merged**